cmd/artifactPrepareVersion_generated.go

```diff
@@ -225,13 +225,18 @@ func artifactPrepareVersionMetadata() config.StepData {
 		},
 		Spec: config.StepSpec{
 			Inputs: config.StepInputs{
+				Secrets: []config.StepSecrets{
+					{Name: "gitHttpsCredentialsId", Description: "Jenkins 'Username with password' credentials ID containing username/password for http access to your git repository.", Type: "jenkins"},
+					{Name: "gitSshKeyCredentialsId", Description: "Jenkins 'SSH Username with private key' credentials ID ssh key for accessing your git repository. You can find details about how to generate an ssh key in the [GitHub documentation](https://docs.github.com/en/enterprise/2.15/user/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent).", Type: "jenkins", Aliases: []config.Alias{{Name: "gitCredentialsId", Deprecated: true}}},
+				},
 				Parameters: []config.StepParameters{
 					{
 						Name:        "buildTool",
 						ResourceRef: []config.ResourceReference{},
 						Scope:       []string{"GENERAL", "PARAMETERS", "STAGES", "STEPS"},
 						Type:        "string",
 						Mandatory:   true,
+						Default:     os.Getenv("PIPER_buildTool"),
```

Comment:
@OliverNocon empty default values are overwritten by [this function](https://github.com/SAP/jenkins-library/blob/master/pkg/generator/helper/helper.go#L470) before entering the templating, which results in `os.Getenv()` calls on parameters that won't be used like that.

Is this the intended usage of the setDefaultParameters function here?

Reply:
Yes, it is intended to allow fetching from env in case there is no default.
This is for example used for secrets.